### ISSUE TITLE
Cross Entropy loss is NaN due to zero element mask at some point

### ISSUE THREAD

**Issue by inspirit** [lucidrains/soundstorm-pytorch#36]
Hi Phil! 
It seems like random mask that is used for predicting masked tokens in cross entropy loss sometimes can be empty (no masked elements) so it results in empty arguments for cross_entropy loss function, we need insure that at least something is masked for we can compute loss?

**Comment by lucidrains** [comment]
@inspirit oh i see, think i built something to ensure at least one being masked but never used it

can you check if the latest commit fixes it?

**Comment by inspirit** [comment]
hmm seems to still happen at some point of time 
it seems like it happens later in the mask definitions

**Comment by inspirit** [comment]
i guess it got cleared during no_replace_prob/random_token_prob logic blocks
https://github.com/lucidrains/soundstorm-pytorch/blob/2b9263e17a1561580928ad2dbd40b0f24e1b2a38/soundstorm_pytorch/soundstorm.py#L1131-L1145

**Comment by inspirit** [comment]
it seem to be fine if we set `random_token_prob = 0.`

**Comment by lucidrains** [comment]
@inspirit one more time?

**Comment by lucidrains** [comment]
@inspirit sorry, i'm unable to test this due to some problem with fairseq

**Comment by inspirit** [comment]
yeah, looks good now! thank you!

**Comment by lucidrains** [comment]
@inspirit thanks for reporting!
